### PR TITLE
Object detection 2d camera demos

### PR DESCRIPTION
**Pull request by ManosMpampis**
This PR contains camera demos for OpenDR's object detection tools in python and C api.
Additionally included is a change to the Nanodet tool that makes it run almost 2 times faster than before.

### CODE REVIEW

**Comment by ManosMpampis** [comment]
This PR is a continuation of the ([#383](https://github.com/opendr-eu/opendr/pull/383)) and ([#352](https://github.com/opendr-eu/opendr/pull/352))

File: docs/reference/c-object-detection-2d-nanodet-jit-h.md [hunk 1]
```diff
@@ -41,8 +41,7 @@ The function returns an OpenDR vector of detections structure with the inference
 
 ### Function *drawBboxes()*
 ```C
-void drawBboxes(OpenDRImageT *image, NanodetModelT *model, OpenDRDetectionVectorTargetT *vector);
+void drawBboxes(OpenDRImageT *image, NanodetModelT *model, OpenDRDetectionVectorTargetT *vector, int show);
 ```
-This function draws the given detections (*vector*) onto the input image (*image*) and then shows the image on screen.
+This function draws the given detections (*vector*) onto the input image (*image*) and then if (*shows*) value is zero, the image will be displayed until a key is pressed.
```
Comment (tsampazk) [inline docs/reference/c-object-detection-2d-nanodet-jit-h.md:46]:
```suggestion
This function draws the given detections (*vector*) onto the input image (*image*) and then if (*show*) value is zero, the image will be displayed until a key is pressed.
```

File: docs/reference/object-detection-2d-nanodet.md [hunk 2]
```diff
@@ -109,15 +109,17 @@ Parameters:
 - **input** : *object*\
   Object of type engine.data.Image.
   Image type object to perform inference on.
-- **threshold**: *float, default=0.35*\
+- **conf_threshold**: *float, default=0.35*\
   Specifies the threshold for object detection inference.
   An object is detected if the confidence of the output is higher than the specified threshold.
+- **iou_threshold**: *float, default=0.6*\
+  Specifies the iou threshold for nms in inference.
```
Comment (tsampazk) [inline docs/reference/object-detection-2d-nanodet.md:116]:
```suggestion
  Specifies the IOU threshold for NMS in inference.
```

File: docs/reference/object-detection-2d-nanodet.md [hunk 3]
```diff
@@ -138,6 +140,11 @@ Parameters:
   Enables the maximum verbosity.
 - **optimization**: *str, default="jit"*\
   It determines what kind of optimization is used, possible values are *jit* or *onnx*.
+- **conf_threshold**: *float, default=0.35*\
+  Specifies the threshold for object detection inference.
+  An object is detected if the confidence of the output is higher than the specified threshold.
+- **iou_threshold**: *float, default=0.6*\
+  Specifies the iou threshold for nms in inference.
```
Comment (tsampazk) [inline docs/reference/object-detection-2d-nanodet.md:147]:
```suggestion
  Specifies the IOU threshold for NMS in inference.
```

**Review by tsampazk — CHANGES_REQUESTED** [review]
Thank you! Some very minor comments.

**Review by tsampazk — APPROVED** [review]
LGTM thank you

**Review by ad-daniel — APPROVED** [review]
I got several false positives (multiple detections) in some of them but I don't think the problem comes from this PR but from the underlying model, I think we can merge it. Thank you

**Comment by ad-daniel** [comment]
The failed test is a network timeout, I think we can merge

**Pull request merged**